File: zerojudge/gcd.py
a=input().split()
def rec(a,b):
    if a==0 or b==0:
        return a+b
    if a>b:
        return rec(a%b,b)
    else:
        return rec(b%a,a)
print(rec(int(a[0]),int(a[1])))

File: zerojudge/gcd.cpp
#include <iostream>
using namespace std;
int rec(int a,int b){
	if (a==0 || b==0)return a+b;
    if (a>b)return rec(a%b,b);
    else return rec(b%a,a);
}
int main(){
	int a,b;
	cin >>a >>b;
	cout <<rec(a,b);
}
